pkg/operators/clustermanager/controllers/migrationcontroller/migration_controller.go

```diff
@@ -174,70 +208,129 @@ func removeStorageVersionMigrations(
+				Status:  metav1.ConditionFalse,
+				Reason:  fmt.Sprintf("StorageVersionMigration Failed. %v", migrationStatusCondition.Reason),
+				Message: fmt.Sprintf("Failed to wait StorageVersionMigration %v succeed. %v", existing.Name, migrationStatusCondition.Message),
+			}, nil
+		case migrationv1alpha1.MigrationRunning:
+			return metav1.Condition{
+				Type:    MigrationSucceeded,
+				Status:  metav1.ConditionFalse,
+				Reason:  fmt.Sprintf("StorageVersionMigration Running. %v", migrationStatusCondition.Reason),
+				Message: fmt.Sprintf("Wait StorageVersionMigration %v succeed. %v", existing.Name, migrationStatusCondition.Message),
+			}, nil
+		}
+	}
+	return metav1.Condition{
+		Type:    MigrationSucceeded,
+		Status:  metav1.ConditionTrue,
+		Reason:  "StorageVersionMigrationSucceed",
+		Message: fmt.Sprintf("All StorageVersionMigrations Succeed"),
+	}, nil
+}
+
 func removeStorageVersionMigration(
 	ctx context.Context,
 	migrationClient migrationv1alpha1client.StorageVersionMigrationsGetter,
 	manifests resourceapply.AssetFunc,
 	file string) error {
-	objectRaw, err := manifests(file)
-	if err != nil {
-		return err
-	}
-	object, _, err := genericCodec.Decode(objectRaw, nil, nil)
+	required, err := parseStorageVersionMigrationFile(manifests, file)
 	if err != nil {
 		return err
 	}
-
-	required, ok := object.(*migrationv1alpha1.StorageVersionMigration)
-	if !ok {
-		return fmt.Errorf("invalid StorageVersionMigration in file %q: %v", file, object)
-	}
-
 	err = migrationClient.StorageVersionMigrations().Delete(ctx, required.Name, metav1.DeleteOptions{})
 	if errors.IsNotFound(err) {
 		return nil
 	}
 	return err
 }
 
-func applyStorageVersionMigration(
-	client migrationv1alpha1client.StorageVersionMigrationsGetter,
+func parseStorageVersionMigrationFile(
 	manifests resourceapply.AssetFunc,
-	recorder events.Recorder,
-	file string) (*migrationv1alpha1.StorageVersionMigration, bool, error) {
+	file string,
+) (*migrationv1alpha1.StorageVersionMigration, error) {
 	objBytes, err := manifests(file)
 	if err != nil {
-		return nil, false, fmt.Errorf("missing %q: %v", file, err)
+		return nil, fmt.Errorf("missing %q: %v", file, err)
 	}
-	requiredObj, _, err := genericCodec.Decode(objBytes, nil, nil)
+	svmObj, _, err := genericCodec.Decode(objBytes, nil, nil)
 	if err != nil {
-		return nil, false, fmt.Errorf("cannot decode %q: %v", file, err)
+		return nil, fmt.Errorf("cannot decode %q: %v", file, err)
 	}
 
-	required, ok := requiredObj.(*migrationv1alpha1.StorageVersionMigration)
+	svm, ok := svmObj.(*migrationv1alpha1.StorageVersionMigration)
 	if !ok {
-		return nil, false, fmt.Errorf("invalid StorageVersionMigration in file %q: %v", file, requiredObj)
+		return nil, fmt.Errorf("invalid StorageVersionMigration in file %q: %v", file, svmObj)
 	}
 
+	return svm, nil
+}
+
+func applyStorageVersionMigration(
+	client migrationv1alpha1client.StorageVersionMigrationsGetter,
+	required *migrationv1alpha1.StorageVersionMigration,
```

Comment:
Seems the required comes from `parseStorageVersionMigrationFile`. Need to check required is nil here?